frontend/simple/views/components/Transitions/MaskToModal/mixinTransitions.js

```diff
@@ -0,0 +1,167 @@
+import Velocity from 'velocity-animate'
+
+/**
+There are 4 components that composes these transitions:
+- the `MaskToModal` - a wrapper around all the other components that keeps the elementsSpecs
+- the `Trigger` - a contribution or the warning message to add income details
+- the `Target` - Income Details modal
+- the `Masker` - A simple blue empty box
+
+The secret of this animation is the `<masker>`:  It's responsible for the major
+part of the animation.
+Here's what happens when the user clicks on a Trigger:
+  1. `Trigger` fades out.
+  2. the `Target`, is added to the DOM _invisible_.
+  3. `Masker` fadesIn at the exactly same time that Trigger is fading out and
+  takes its specs (updateSpecsOf).
+  4. `Masker` animates to take the `Target` dimensions/position
+  5. Once that animation is finished `Masker` fades out and `Target` fades in at the same time.
+
+When closing the Target the same animation happens in the inverted order but slightly faster.
+
+Entrance animations are designed to be smoother than exits so the user better
+understands the context of the new content easier (less cognitive process).
+The exit time is faster than the entrance time because the user already knows
+the original context.
+If both entrance/exit animations take the same time the user will percive the exit animation
+as being slower that the entrance, this is a way to trick his/her mind.
+This time is cutted at the Trigger fadeOut moment.
+*/
+
+const animationMixins = {
+  data () {
+    return {
+      config: {
+        // Trigger/Target time to fadeIn/Out - give a delay, so the Masker starts first.
+        fade: 75,
+        fadeDelay: 50,
+        // Target/Masker time to fade back to original position - slightly faster
+        fadeBack: 50,
+        // Masker time to animate between the Target and a Trigger (and vice versa)
+        maskerTime: 250,
+        // The time for Target/Trigger to fadeIn + maskerTime
+        delay: 300
```

Comment:
Is this supposed to be 300 or 325, and would it be better to compute this value using variables based on the other two instead of manually?

Reply:
Well, another trick about motion design, is that it's not all about perfect math. Sometimes numbers need to be manually adjusted to the eye perception, otherwise it might not feel so natural... in this case the fadeIn/out between the mask and the original element seemed to happen too late, specially on devices with slower CPU. There's a important variable to have in mind when doing real time motion - the machine CPU - that can delay things a little. I did all these tests with devTools Performance to verify each frame and those 25/50ms did the difference...